packages/compass-crud/src/stores/crud-store.spec.ts

```diff
@@ -117,8 +111,9 @@ function waitForState(store, cb, timeout) {
 }
 
 describe('store', function () {
-  this.timeout(5000);
+  this.timeout(30_000);
```

Comment:
This was here since forever (weirdly enough it's smaller than our default timeout), so not really part of the PR that needs to be addressed, but maybe would make sense to just define this once in .mocharc file? Seems like we set it to 30_000 in a few test files

Reply:
Fwiw, I bumped it up to 2 minutes here because the server download in CI takes a while (which, yes, probably fair to mention as a disadvantage of this model).

I’m okay with adding it to a shared rc file, yes :+1:

Reply:
Yeah, I get that, I think it's okay, this happens anyway, just in the pretest script currently, I think it's fine if we bump the timeouts for this inside the tests.

I was actually thinking more like adding this to the .mongoshrc file in compass-crud, but yeah, I guess if this will apply to a lot of test suites as probably a lot of them use server it might make sense to even stick it in the shared one. Alternatively maybe this is something that the mocha helper that you added can do? This is kinda going on a tangent, but looking at how this is used in the tests and from some common mocha patterns, maybe it can be changed a little to work sort of like this:

```ts
// mocha helper
export funciton setupTestServer(opts) {
  // setting high timeout because we are anticipating that starting cluster from
  // tests will take more time than our default timeout allows for
  this.timeout(60_000);
  beforeEach(async function() {
    this.cluster = await startTestServer(opts);
  });
  afterEach(async function () {
    if (opts.alwaysStartNewServer) await cluster.stop();
  })
  after(async function () {
    if (!opts.alwaysStartNewServer) await cluster.stop();
  })
}

// some-sort-of-test.spec.ts
describe('when connected to mongodb', function() {
  setupTestServer();

  let client

  before(async function () {
    client = new MongoClient(this.cluster.connectionString);
    await client.connect();
  })
})
```

That way all the test server logic that we have to repeat every time stays in one place, so chances of forgetting to add a timeout or forget the clean up are nonexistent and we can also document things like increased timeout times in one place

Reply:
(no strong preference though, just a suggestion!)

Reply:
Yeah no, this makes sense, and it’s ultimately what we ended up doing in mongosh as well :+1: 04b46470dd8eebe92b48d968cc19bfcd49a8eba4 contains this.